[0.9.0] doesn't allow disabled systemd

With version 0.9.9, despite `--with-initscript=none`, the build errors out with

```
configure: error: conditional "HAVE_SYSTEMD_DAEMON" was never defined.
Usually this means the macro was only invoked conditionally.
```

I think this commit is responsible, but haven't tracked it down further.
Might be the m4 systemd macro.

https://github.com/gssapi/gssproxy/commit/145c7cecb9d7aecb707c02995b0289b40a6df4d7

--- a/configure.ac
+++ b/configure.ac
@@ -164,13 +164,13 @@ if test x$only_gss_module != xyes; then
                          [AC_MSG_ERROR([Could not find libcap headers])])
     fi
 
-    if test x$initscript = xsystemd; then
-        WITH_SYSTEMD_UNIT_DIR
-        WITH_SYSTEMD_USER_UNIT_DIR
-        AM_CHECK_SYSTEMD
-    fi
 fi
 AM_CONDITIONAL([BUILD_PROXY], [test x$only_gss_module != xyes ])
+if test x$initscript = xsystemd; then
+    WITH_SYSTEMD_UNIT_DIR
+    WITH_SYSTEMD_USER_UNIT_DIR
+fi
+AM_CHECK_SYSTEMD
 
 AC_ARG_ENABLE([always-interpose],
               [AC_HELP_STRING([--enable-always-interpose],

